es5-shim.js

```diff
@@ -175,11 +176,143 @@ var ES = {
+
+if (shouldPatchCallApply) {
+    // To prevent recursion when `call` and `apply` are patched. Robustness.
+    call.call = call;
+    call.apply = apply;
+    apply.call = call;
+    apply.apply = apply;
+}
+
+// This function is for use within `call` and `apply` only.
+// To avoid any possibility of `call` recursion we use original `hasOwnProperty`.
+var isDuckTypeArguments = hasToStringTagLegacyArguments && (function (hasOwnProperty) {
+  return function (value) {
+      if (value != null) { // Checks `null` or `undefined`.
+          if (typeof value === 'object' && call.call(hasOwnProperty, value, 'length')) {
+              var length = value.length;
+              // Constant. ES3 maximum array length. 2^32-1
+              if (length > -1 && length % 1 === 0 && length <= 4294967295) {
+                  return !call.call(hasOwnProperty, value, 'arguments') && call.call(hasOwnProperty, value, 'callee');
+              }
+          }
+      }
+      return false;
+  };
+}(ObjectPrototype.hasOwnProperty));
+
+// For use with `call` and `apply` fixes.
+var toStringTag = shouldPatchCallApply && function (value) {
+    // Add whatever fixes for getting `[[Class]]` strings here.
+    if (value === null) {
+        return '[object Null]';
+    }
+    if (typeof value === 'undefined') {
+        return '[object Undefined]';
+    }
+    if (hasToStringTagLegacyArguments && isDuckTypeArguments(value)) {
+        return '[object Arguments]';
+    }
+    // `to_string` is safe to use here, no known issue at present.
+    return call.call(to_string, value);
+};
+
+defineProperties(FunctionPrototype, {
+    // ES-5 15.3.4.3
+    // http://es5.github.io/#x15.3.4.3
+    // The apply() method calls a function with a given this value and arguments
+    // provided as an array (or an array-like object).
+    apply: function (thisArg) {
+        var argsArray = arguments[1];
+        var type = typeof argsArray;
+        if (arguments.length > 1) {
+            // IE9 (though fix not needed) has a problem here for some reason!!!
+            // Pretty much any function here causes error `SCRIPT5007: Object expected`.
+            if (type !== 'undefined' && type !== 'object' && type !== 'function') {
+                throw new TypeError('Function.prototype.apply: Arguments list has wrong type');
+            }
+        }
+        // If `this` is `Object#toString`, captured or modified.
+        if (this === to_string || this === Object.prototype.toString) {
+            return toStringTag(thisArg);
+        }
+        // All other applys.
+        if (arguments.length > 1 && type === 'object' && argsArray && argsArray.length > 0) {
+            // Boxed string access bug fix.
+            if (splitString && to_string.call(argsArray) === '[object String]') {
+                // `str_split` is safe to use here, no known issue at present.
+                argsArray = call.call(str_split, argsArray, '');
```

Comment:
there's already a call-bound `strSplit` btw, you should just be able to do `strSplit(argsArray, '')`